### ISSUE TITLE
Size of LaTeX legends cannot be calculated

### ISSUE THREAD

**Issue by heliosdrm** [jheinen/GR.jl#317]
Example:
```julia
julia> using LaTeXStrings
julia> x = range(0, 1, length=100);
julia> plot(x, sqrt.(x), labels=(L"\sqrt{x}",))
sh: 1: dvipng: not found
dvipng: PNG conversion failed
sh: 1: dvipng: not found
dvipng: PNG conversion failed
```
(The legend is properly drawn, but the box does not fit the legend.)

The underlying reason seems to be that `GR.inqmathtex` does not work (it doesn't matter what string you use):

```julia
julia> GR.inqmathtex(0, 0, "\\sqrt{x}")
sh: 1: dvipng: not found
dvipng: PNG conversion failed
([0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0])
```

This is using GR 0.50.0, in Ubuntu 16.04.

**Comment by jheinen** [comment]
Aee you really using the latest run-time? `ENV["GRDIR"]=""; ] build GR`
<img width="712" alt="Screen Shot 2020-05-31 at 07 29 50" src="https://user-images.githubusercontent.com/5718244/83345191-97cfbe80-a310-11ea-9995-60c9c6a76299.png">

**Comment by heliosdrm** [comment]
Yes, I did it. (And tried again, just in case, but with same results).

![imagen](https://user-images.githubusercontent.com/10651028/83347994-4d0e7080-a329-11ea-9b47-16afcd48d137.png)

The legend is displayed, i.e. `GR.mathtex` is working properly, it is `GR.inqmathtex` what seems to be broken in my system. I don't have a system-wide installation of LaTeX, I don't know if that may have something to do.

**Comment by heliosdrm** [comment]
For the record, I see the same problem in Windows 10, although the error message of `inqmathex` is different:

```julia
julia> GR.inqmathtex(0,0,"\$abc\$")
"latex" no se reconoce como un comando interno o externo,
programa o archivo por lotes ejecutable.
latex: failed to create a dvi file
([0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0])
```
(translation from Spanish: `"latex" is not recognized as an internal or external command, program or executable batch file`)

(Also, I have executed `ENV["GRDIR"]=""; ] build GR` before `using GR`)

**Comment by jheinen** [comment]
... and `version()` is `"0.50.0"`?

**Comment by heliosdrm** [comment]
This is a verbatim output of a session in Windows 10 (later I will copy it for Ubuntu) - I don't have OSX

```julia
julia> ENV["GRDIR"] = ""
""

(testGR) pkg> build GR
  Building GR → `D:\.julia\packages\GR\QXeZO\deps\build.log`

julia> using GR
[ Info: Precompiling GR [28b8d3ca-fb5f-59d9-8090-bfdbd6d07a71]

julia> GR.version()
"0.50.0.post1"

julia> x = range(0,1,length=100);

julia> GR.inqmathtex(0,0,"\$abc\$")
"latex" no se reconoce como un comando interno o externo,
programa o archivo por lotes ejecutable.
latex: failed to create a dvi file
([0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0])
```

(And the problem with the legend box size is the same as reported above.)

**Comment by heliosdrm** [comment]
And in Ubuntu 16.04:

```julia
julia> ENV["GRDIR"] = ""
""

(testGR) pkg> build GR
  Building MbedTLS → `~/.julia/packages/MbedTLS/X4xar/deps/build.log`
  Building GR ─────→ `~/.julia/packages/GR/QXeZO/deps/build.log`

julia> using GR
[ Info: Recompiling stale cache file /home/meliana/.julia/compiled/v1.1/GR/NDU5Y.ji for GR [28b8d3ca-fb5f-59d9-8090-bfdbd6d07a71]

julia> GR.version()
"0.50.0.post1"

julia> x = range(0,1,length=100);

julia> GR.inqmathtex(0,0,"\$abc\$")
sh: 1: dvipng: not found
dvipng: PNG conversion failed
([0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0])
```

Might the Julia version have anything to do? I'm using 1.3, not the latest one.

**Comment by heliosdrm** [comment]
In this issue you recommended install `texlive-dvipng`. That was before version 0.50.
https://github.com/jheinen/GR.jl/issues/93#issuecomment-451855617

Is that recommendation still valid, or was 0.50 meant to fix it without external installation of `dvipng`?

**Comment by jheinen** [comment]
Ah, now I see. Right now, it only works with the convenience functions:
<img width="1331" alt="font-test" src="https://user-images.githubusercontent.com/5718244/83357575-689b6a80-a36d-11ea-8990-844040e80ba3.png">


Will have to fix it for the low-level functions ...

**Comment by heliosdrm** [comment]
I'd say that it's not really a matter of "convenience functions" (i.e. `jlgr`) vs "low-level" ones (`GR`). As commented before, `GR.mathtex` seems to work fine; it is specifically `GR.inqmathtex` what I see failing.

Most convenience functions and attributes (`title`, `x/y/zlabel`, etc.) work with LaTeX expressions without problem, because they are made through `GR.mathtex` (via `jlgr.text`). But legends may be buggy, because `legend_size` and `draw_legend` also use `GR.inqmathtex` (via `jlgr.inqtext`).

**Comment by skleinbo** [comment]
For me there is a difference in Windows vs. MacOS:

_Windows 10:_
```julia
julia> GR.inqmathtex(0,0,L"\alpha")
([0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0])
```
which results in latex labels overlapping with the axis when plotting with Plots.jl

**Comment by jheinen** [comment]
And there is no error messag, like `latex: failed to create a dvi file`?

Could you please try:
```
import GR
using LaTeXStrings
GR.settextfontprec(232, 3) # use GR's math renderer
GR.inqmathtex(0,0, L"\alpha")
```
Should give something like:
```
([0.0, 0.023753611246744792, 0.023753611246744792, 0.0], [0.0, 0.0, 0.016803385416666667, 0.016803385416666667])
```

**Comment by skleinbo** [comment]
Funny thing. The `failed to create[...]` messages popped up when I quit the REPL 😕 

Again, those commands give output similar to what you wrote on Mac, but result in only zeroes on Windows.

The TeXLive `bin` directory is in my PATH if that makes a difference. Though I understand GR should use its own renderer instead?

**Comment by heliosdrm** [comment]
I have tested today, with the version in master. Now it works fine in Linux. :+1: 

However there are still problems in Windows, but not the same as before. I'll investigate a bit further.

**Comment by heliosdrm** [comment]
Windows issue fixed with GR v0.51. Great!
